rhizome/host/lib/slice_setup.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require_relative "../../common/lib/util"
+require "fileutils"
+
+class SliceSetup
+  def initialize(slice_name)
+    @slice_name = slice_name
+  end
+
+  def systemd_service
+    @systemd_service ||= File.join("/etc/systemd/system", @slice_name)
+  end
+
+  def prep(allowed_cpus)
+    install_systemd_unit(allowed_cpus)
+    start_systemd_unit
+  end
+
+  def purge
+    if File.exist? systemd_service
+      r "systemctl stop #{@slice_name.shellescape}"
+      FileUtils.rm_f(systemd_service)
+
```

Comment:
Perhaps this conditional can be made more robust. It tries to get something like `rm -f` semantics is how I'm reading it. It's the same as the anti-pattern of "File.exist?; File.rm", where it's almost always better to handle the rm exception.

In this case, I would try to stop the slice and absorb exit code 5, the "not found" exit status. Perhaps `r` can be expanded to support acceptable exit codes to avoid exception rigmarole, e.g. `expect: [0, 5]` as a keyword argument. The default would be `expect: 0` . Other nomenclature welcome, because `expect` looks a bit too much like rspec for comfort.

Another thing that has borrowed me is that `FileUtils.rm_f` (and indeed `rm -f`) is too relaxed about things like EPERM, which would be a reason to crash. We really only want to absorb ENOENT. Perhaps that can be a `common/lib/util` routine.

Reply:
done